magma/backend/mlir/mlir.py

```diff
@@ -197,6 +197,15 @@ def new_region(self) -> MlirRegion:
     def set_parent(self, parent: MlirBlock):
         self.parent = weakref.ref(parent)
 
+    def parent_op(self) -> Optional['MlirOp']:
+        block = maybe_dereference(self.parent)
+        if block is None:
+            return None
+        region = maybe_dereference(block.parent)
+        if region is None:
+            return None
+        return maybe_dereference(region.parent)
```

Comment:
Since we're only testing Python 3.8 and above, perhaps we can start using the walrus operator
```suggestion
        if not (block := maybe_dereference(self.parent)):
            return None
        if not (region := maybe_dereference(block.parent)):
            return None
        return maybe_dereference(region.parent)
```